cpp/include/cudf/detail/utilities/host_vector.hpp

```diff
@@ -102,11 +98,12 @@ class rmm_host_allocator {
 #ifdef __CUDACC__
 #pragma nv_exec_check_disable
 #endif
-  template <class... Properties>
-  rmm_host_allocator(async_host_resource_ref<Properties...> _mr, rmm::cuda_stream_view _stream)
+  template <typename ResourceType>
+  rmm_host_allocator(ResourceType _mr, rmm::cuda_stream_view _stream)
     : mr(_mr),
       stream(_stream),
-      _is_device_accessible{contains_property<cuda::mr::device_accessible, Properties...>}
+      _is_device_accessible{
```

Comment:
The existing code has inconsistent naming: `_` prefixes are used for constructor arguments, but `_is_device_accessible` is a data member with a `_` prefix, unlike other members such as `mr` and `stream`.